Bottom: f9af2618a5c79a2efa3100258166734f07f652f4
Top:    84d0745a024d7009a9b60d8ea6751c6abf1e3262
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-14 02:36:41 +0000

Tag circular contours so they can be drawn quicker


---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 1c7ead9..1661526 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -183,6 +183,14 @@ void common_fill_pcb_polygon (hidGC gc, PolygonType *poly,
 static int thindraw_hole_cb (PLINE *pl, void *user_data)
 {
   hidGC gc = user_data;
+
+  /* If the contour is round, use an arc drawing routine. */
+  if (pl->is_round) {
+    gui->set_line_width (gc, 0);
+    gui->draw_arc (gc, pl->cx, pl->cy, pl->radius, pl->radius, 0, 360);
+    return 0;
+  }
+
   thindraw_contour (gc, pl);
   return 0;
 }
